src/battle_ai_main.c

```diff
@@ -1875,6 +1875,8 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 ADJUST_SCORE(-10);
             break;
         case EFFECT_DESTINY_BOND:
+            if (DoesDestinyBondFail(battlerAtk))
+                ADJUST_SCORE(-10);
             if (gBattleMons[battlerDef].status2 & STATUS2_DESTINY_BOND)
                 ADJUST_SCORE(-10);
```

Comment:
Does it make sense to also put the status bit check in the function itself? It is kinda what the function is doing according by the name

Reply:
Sorry, I completely missed the point that the function is also used by the battle engine.